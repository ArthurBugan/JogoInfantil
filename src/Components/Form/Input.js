import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { Field } from 'react-final-form';
import { Input as InputBase, Item, Text, Label } from 'native-base';
import * as validation from './validation';

const renderInput = ({ input, label, placeholder, type, meta, style, styleErro, rounded }) => {
  const password = type === 'password' || type === 'confirmPassword';

  let styleItem = style;
  if (meta.error && meta.touched) {
    styleItem = { ...style, borderColor: '#F00' };
  }
  if (rounded) {
    return (
      <View>
        <Item rounded style={styleItem}>
          <InputBase {...input} secureTextEntry={password} placeholder={placeholder} />
        </Item>
        {meta.error && meta.touched && <Text style={styleErro}>{meta.error}</Text>}
      </View>
    );
  }
  return (
    <View>
      <Item floatingLabel style={styleItem}>
        <Label style={{ marginTop: 10 }}>{label}</Label>
        <InputBase {...input} secureTextEntry={password} />
      </Item>
      {meta.error && meta.touched && <Text style={styleErro}>{meta.error}</Text>}
    </View>
  );
};


const Input = (props) => {
  const { type, validate, required, confirm } = props;
  let validationInternal;
  if (type === 'confirmPassword') {
    validationInternal = validation.confirmPassword2(confirm);
  } else if (typeof validation[type] === 'function' || validate) {
    if (typeof validation[type] === 'function') {
      validationInternal = validation[type];
    } else if (validate) {
      validationInternal = validate;
    }
  }

  if (validationInternal && required) {
    validationInternal = validation.composeValidators(validationInternal, validation.required);
  } else if (!validationInternal && required) {
    validationInternal = validation.required;
  }
  return (
    <Field
      {...props}
      component={renderInput}
      validate={validationInternal}
    />
  );
};

Input.propTypes = {
  type: PropTypes.string.isRequired,
  confirm: PropTypes.string,
  validate: PropTypes.func,
  required: PropTypes.bool,
};
Input.defaultProps = {
  validate: null,
  confirm: null,
  required: false,
};

renderInput.propTypes = {
  input: PropTypes.object.isRequired,
  style: PropTypes.object,
  styleErro: PropTypes.object,
  meta: PropTypes.object,
  type: PropTypes.string.isRequired,
  label: PropTypes.string,
  rounded: PropTypes.bool,
  placeholder: PropTypes.string,
};
renderInput.defaultProps = {
  style: {},
  styleErro: { color: '#F00', fontSize: 13, textAlign: 'center', marginTop: -17 },
  meta: {},
  rounded: false,
  label: null,
  placeholder: null,
};
export default Input;
