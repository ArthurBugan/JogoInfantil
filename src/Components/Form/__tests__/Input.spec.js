import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { expect as chaiEx } from 'chai';
import Adapter from 'enzyme-adapter-react-16';
import Input from '../Input';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
describe('Test Props', () => {
  beforeEach(() => {
    wrapper = shallow(<Input
      name="email"
      label="E-mail"
      type="email"
      placeholder="E-mail"
      rounded
      required
    />);
  });
  it('name = email ', () => {
    chaiEx(wrapper.prop('name')).to.equals('email');
  });
  it('label = E-mail ', () => {
    chaiEx(wrapper.prop('label')).to.equals('E-mail');
  });
  it('type = email ', () => {
    chaiEx(wrapper.prop('type')).to.equals('email');
  });
  it('placeholder = E-mail ', () => {
    chaiEx(wrapper.prop('placeholder')).to.equals('E-mail');
  });
  it('rounded = True ', () => {
    chaiEx(wrapper.prop('rounded')).to.ok;
  });
  it('required = True ', () => {
    chaiEx(wrapper.prop('required')).to.ok;
    expect(wrapper).toMatchSnapshot();
  });
});
