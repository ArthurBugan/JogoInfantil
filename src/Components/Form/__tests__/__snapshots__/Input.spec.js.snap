// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Props required = True  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    label="E-mail"
    name="email"
    placeholder="E-mail"
    required={true}
    rounded={true}
    type="email"
    validate={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "component": [Function],
      "format": [Function],
      "label": "E-mail",
      "name": "email",
      "parse": [Function],
      "placeholder": "E-mail",
      "required": true,
      "rounded": true,
      "type": "email",
      "validate": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "component": [Function],
        "format": [Function],
        "label": "E-mail",
        "name": "email",
        "parse": [Function],
        "placeholder": "E-mail",
        "required": true,
        "rounded": true,
        "type": "email",
        "validate": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
